### PR TITLE
Move the software version to resource page.

### PR DESCRIPTION
**Pull request by poormusic2001**
1. Move software version functions.
2. Add the db access condition with resource.

### CODE REVIEW

**Comment by Echolon** [comment]
Thank you Jim @poormusic2001 

There is a build error unfortunately 🙁 Lets see reviewers will say

File: Monal/Classes/DataLayer.m [hunk 1]
```diff
@@ -625,11 +625,11 @@ -(NSArray*) resourcesForContact:(NSString*) contact
     return resources;
 }
 
--(NSArray*) softwareVersionInfoForAccount:(NSString*)account andContact:(NSString*)contact
+-(NSArray*) softwareVersionInfoForAccount:(NSString*)account contact:(NSString*)contact andResource:(NSString*)resource
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/DataLayer.m:628]: I don't like that method name
`getSoftwareVersionInfoForContact:andAccount:` would be better.

File: Monal/Classes/DataLayer.h [hunk 2]
```diff
@@ -61,8 +61,8 @@ extern NSString *const kMessageTypeUrl;
 
 -(NSMutableArray*) onlineContactsSortedBy:(NSString*) sort;
 -(NSArray*) resourcesForContact:(NSString*)contact ;
--(NSArray*) softwareVersionInfoForAccount:(NSString*)account andContact:(NSString*)contact;
--(void) setSoftwareVersionInfoForAppName:(NSString*)appName appVersion:(NSString*)appVersion platformOS:(NSString*)platformOS withAccount:(NSString*)account andContact:(NSString*)contact;
+-(NSArray*) softwareVersionInfoForAccount:(NSString*)account contact:(NSString*)contact andResource:(NSString*)resource;
+-(void) setSoftwareVersionInfoForAppName:(NSString*)appName appVersion:(NSString*)appVersion platformOS:(NSString*)platformOS withAccount:(NSString*)account contact:(NSString*)contact andResource:(NSString*)resource;
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/DataLayer.h:65]: I don't like this method name.
`setSoftwareVersionInfoForContact:resource:andAccount:withAppName:appVersion:andPlatformOS:` would be much better.

File: Monal/Classes/MLIQProcessor.m [hunk 3]
```diff
@@ -463,13 +463,15 @@ +(void) iqVersionResult:(XMPPIQ*) iqNode forAccount:(xmpp*) account
...
-                                                             andContact:iqNode.fromUser];
+                                                                contact:iqNode.fromUser
+                                                            andResource:iqNode.fromResource];
             
             [[NSNotificationCenter defaultCenter] postNotificationName:kMonalXmppUserSoftWareVersionRefresh
                                                                 object:account
-                                                              userInfo:@{@"platform_App_Name":iqAppName,
-                                                                      @"platform_App_Version":iqAppVersion,
-                                                                               @"platform_OS":iqPlatformOS}];
+                                                              userInfo:@{iqNode.fromResource:@{
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLIQProcessor.m:471]: Don't use something that could be nil as key.
Just set the `fromResource` as one extra key in your `userInfo` dictionary.

File: Monal/Classes/MLXMPPManager.m [hunk 4]
```diff
@@ -793,20 +793,15 @@ -(void) getVCard:(MLContact *) contact
     [account getVCard:contact.contactJid];
 }
 
--(void) getEntitySoftWareVersion:(MLContact *) contact
+-(void) getEntitySoftWareVersion:(MLContact *) contact andResource:(NSString*) resource
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLXMPPManager.m:796]: name again: `getEntitySoftWareVersionForContact:andResource:`

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]
See my inline comments

**Comment by tmolitor-stud-tu** [comment]
@Echolon the build failed because our tests used in the CI are not yet updated to the new parser.

**Pull request merged**